src/Microsoft.AspNetCore.Razor.Language/DefaultRazorDiagnostic.cs

```diff
@@ -8,26 +8,29 @@ namespace Microsoft.AspNetCore.Razor.Language
 {
     internal class DefaultRazorDiagnostic : RazorDiagnostic
     {
-        private readonly RazorDiagnosticDescriptor _descriptor;
-        private readonly object[] _args;
-
         internal DefaultRazorDiagnostic(RazorDiagnosticDescriptor descriptor, SourceSpan span, object[] args)
         {
-            _descriptor = descriptor;
+            Descriptor = descriptor;
             Span = span;
-            _args = args;
+            Args = args;
         }
 
-        public override string Id => _descriptor.Id;
+        public override string Id => Descriptor.Id;
 
-        public override RazorDiagnosticSeverity Severity => _descriptor.Severity;
+        public override RazorDiagnosticSeverity Severity => Descriptor.Severity;
 
         public override SourceSpan Span { get; }
 
+        // Internal for testing
+        internal RazorDiagnosticDescriptor Descriptor { get; }
```

Comment:
Exposed a little more information here for tests. Basically in the tests we need to copy diagnostics and update the diagnostics location. Hate that the tests were written that way but it's more code than I'm willing to invest the time in moving.